corehq/apps/app_manager/views/modules.py

```diff
@@ -1009,36 +1009,47 @@ def _update_search_properties(module, search_properties, lang='en'):
     True
 
     """
+
+    # Replace translation for current language in a translations dict
+    def _update_translation(old_obj, new_data, old_attr, new_attr=None):
```

Comment:
At it's core this is a way to join together the old and new dicts, so you could limit the scope of the function to just that - make it accept the old dict and the new translation, then handle the `pop`ing and joining here, but leave figuring out how to find the old dict and the new translation to the callers.  I am mildly annoyed by the need to guard against `current = None` everywhere, though.  Oh well, maybe there's not a perfectly clean option here.